Sources/SwiftParser/Parser.swift

```diff
@@ -122,6 +127,18 @@ public struct Parser: TokenConsumer {
   @_spi(RawSyntax)
   public var currentToken: Lexer.Lexeme
 
+  /// The current nesting level, i.e. the number of tokens that
+  /// `startNestingLevel` minus the number of tokens that `endNestingLevel`
+  /// which have been consumed so far.
+  public var nestingLevel: Int = 0
+
+  /// When this nesting level is exceeded, the parser should stop parsing.
+  public let maximumNestingLevel: Int
+
+  /// A default maximum nesting level that is used if the client didn't
+  /// explicitly specify one.
+  public static let defaultMaximumNestingLevel = 256
```

Comment:
I know it's terrible, but should we do some kind of `#if DEBUG` check to decide whether to use the lower limit or higher limit?

Reply:
That’s actually a good idea. The entire check is pretty horrible in itself, so a `#if DEBUG` check doesn’t make a huge difference anymore.